drivers/mdio/mdio_nxp_enet_qos.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_enet_qos_mdio
+
+#include <zephyr/net/mdio.h>
+#include <zephyr/drivers/mdio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/ethernet/eth_nxp_enet_qos.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mdio_nxp_enet_qos, CONFIG_MDIO_LOG_LEVEL);
+
+struct nxp_enet_qos_mdio_config {
+	const struct device *enet_dev;
+};
+
+struct nxp_enet_qos_mdio_data {
+	struct k_mutex mdio_mutex;
+};
+
+struct mdio_transaction {
+	enum mdio_opcode op;
+	union {
+		uint16_t write_data;
+		uint16_t *read_data;
+	};
+	uint8_t portaddr;
+	uint8_t regaddr;
+	enet_qos_t *base;
+	struct k_mutex *mdio_bus_mutex;
+};
+
+static bool check_busy(enet_qos_t *base)
```

Comment:
Should we `inline` this function?

Reply:
I don't think you should ever inline functions in a C file. Let the compiler decide to inline if it helps.